### ISSUE TITLE
Question about reproduction

### ISSUE THREAD

**Issue by lc-nyovchev** [reznok/Spring4Shell-POC#2]
Hey @reznok, awesome work!

I was wondering if having the `Model` and `@ModelAttribute` are required to be susceptible to the exploit? Often, a Controller is annotated with `@RestController` and looks more like:

```java
@RestController
@RequiredArgsConstructor
public class MyRestController {
    
    @NonNull private final MyService myService;
 
    @GetMapping(value = REQUEST_PREFIX + "/path/{somePathVariable}/items")
    public List<SomePojo> someGetRequest(@PathVariable Integer somePathVariable, @RequestParam(value = "itemIds") Set<UUID> itemIds) {
	    return myService.getSomePojos(somePathVariable, itemIds);
    }
}
```

my question in short is:

Is this exploitable `@RestController` with `@PathVariable`, `@RequestParam`? Or only for Model and `@ModelAttribute` as it's in your example repo?

**Comment by dagnelies** [comment]
As far as I can tell, the vulnerable part is the `@ModelAttribute` in the following method.

    @PostMapping("/greeting")
    public String greetingSubmit(@ModelAttribute Greeting greeting, Model model) {

I've never used `@ModelAttribute` for parameters and it sounds quite strange to me to do so.

Here are the variations I tried:

- removing `@ModelAttribute` => exploit fails
- replacing `@ModelAttribute` with `@RequestParam` => exploit fails
- replacing `@ModelAttribute` with `@RequestParam` and `@Controller` with `@RestController` => exploit fails

...so, please do not misunderstand me. I have no idea if the exploit could be adapted for such circumstances. I just say that "out of the box" this exploit stops working in without that `@ModelAttribute`.


PS: apparently, it's already fixed: https://spring.io/blog/2022/03/31/spring-framework-rce-early-announcement

**Comment by lc-nyovchev** [comment]
Thanks for checking @dagnelies, I had the same experience as well! I couldn't out of the box reproduce the exploit with `@RequestParam`.

In any case, since [Spring released 5.3.18](https://spring.io/blog/2022/03/31/spring-framework-rce-early-announcement), even that exploit should be natively patched (without having to do the global `@InitBinder` `@ControllerAdvice`) and my team already moved to that. Anyway, many many thanks for your answer!
